### PR TITLE
optuna 3.2.0 ❄️

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/optuna/optuna/releases
License: https://github.com/optuna/optuna/blob/v3.2.0/LICENSE
Requirements: https://github.com/optuna/optuna/blob/v3.2.0/pyproject.toml

Actions:
1. Clean up the recipe: 
  - Remove `noarch python`
  - Skip `py<36`
  - Add flags `--no-deps --no-build-isolation` to `script`
  - Add missing `setuptools` & `wheel` to `host`
  - Fix `python` in `host` and `run`
  - Add `pip check`
2. Sort runtime dependencies
3. Add `description`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,30 +11,35 @@ source:
 
 build:
   number: 0
+  skip: True  # [py<37]
```
Comment (boldorider4) [inline recipe/meta.yaml:14]: why this? conda-forge?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,30 +11,35 @@ source:
...
+    - wheel
   run:
     - alembic >=1.5.0
     - cmaes >=0.9.1
     - colorlog
     - numpy
     - packaging >=20.0
-    - python >=3.7
+    - python
+    - PyYAML
```
Comment (boldorider4) [inline recipe/meta.yaml:32]: I like all lower case for consistency....not sure if we have policies for that though

Comment (skupr-anaconda) [reply recipe/meta.yaml:32]: Fixed

**Review by boldorider4 — APPROVED** [review]

**Review by lorepirri — APPROVED** [review]
LGTM!

**Pull request merged**